daprdocs/content/en/operations/performance-and-scalability/perf-service-invocation.md

```diff
@@ -0,0 +1,89 @@
+
+### Kubernetes components
+
+* Sidecar (data plane)
+* Sentry (optional, control plane)
+* Placement (optional, control planee)
+* Operator (control plane)
+* Sidecar Injector (control plane)
+
+For more information see [overview of Dapr on Kubernetes]({{< ref kubernetes-overview.md >}}).
+
+## Performance summary for Dapr v0.11.3
+
+The service invocation API is a reverse proxy with built-in service discovery to connect to other services. This includes tracing, metrics, mTLS for in-transit encryption of traffic, together with resiliency in the form of retries for network partitions and connection errors.
+
+Using service invocation you can call from HTTP to HTTP, HTTP to gRPC, gRPC to HTTP, and gRPC to gRPC. Dapr does not use HTTP for the communication between sidecars, always using gRPC, while carrying over the semantics of the protocol used when called from the app. Service invocation is the underlying mechanism of communicating with Dapr Actors.
+
+For more information see [service invocation overview]({{< ref service-invocation-overview.md >}}).
+
+### Kubernetes performance test setup
+
+The test was conducted on a 3 node Kubernetes cluster, using commodity hardware running 4 cores and 8GB of RAM, without any network acceleration.
+The setup included a load tester ([Fortio](https://github.com/fortio/fortio)) pod with a Dapr sidecar injected into it that called the service invocation API to reach a pod on a different node.
+
+Test parameters:
+
+* 1000 requests per second
+* Sidecar limited to 0.5 vCPU
+* Sidecar mTLS enabled
+* Sidecar telemetry enabled (tracing with a sampling rate of 0.1)
+* Payload of 1KB
+
+The baseline test included direct, non-encrypted traffic, without telemetry, directly from the load tester to the target app.
+
+### Control plane performance
+
+The Dapr control plane uses a total of 0.01 vCPU and 77 Mb.
+
+| Component  | vCPU | Memory
+| ------------- | ------------- | -------------
+| Operator  | 0.001  | 12.5 Mb
+| Sentry  | 0.005  | 13.6 Mb
+| Sidecar Injector  | 0.002  | 14.6 Mb
+| Placement | 0.001  | 20.9 Mb
+
+There are a number of variants that affect the CPU and memory consumption for each of the system components. These variants are shown in the table below.
+
+| Component  | vCPU | Memory
+| ------------- | ------------- | ------------------------
+| Operator  | Number of pods requesting components, configurations and subscriptions  |
+| Sentry  | Number of certificate requests  |
+| Sidecar Injector | Number of admission requests |
+| Placement | Number of actor rebalancing operations | Number of connected actor hosts
+
+### Data plane performance
+
+The Dapr sidecar uses 0.48 vCPU and 23Mb per 1000 requests per second.
+End-to-end, the Dapr sidecar adds 1.57 ms to the 90th percentile latency, and 2.36 ms to the 99th percentile latency. End-to-end here is a call from one app to another app receiving a response. This is shown by steps 1-7 in [this diagram]({{< ref service-invocation-overview.md >}}).
```

Comment:
Is this per each side-car? So if I call from one app to another there are two side cars and double these numbers?

Reply:
No, this is end to end for both sidecars.

Reply:
I updated the text to reflect that.

Reply:
Also can you go to this link to the end of the doc on service invocation "service-invocation-overview" 

* See the [service invocation performance]({{< ref service_invocation_performance.md >}}) numbers

This doc
https://github.com/dapr/docs/blob/master/daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-overview.md

Reply:
At the start you mention self hosted + Kubernetes, but only give number for K8s. Is this intentional?

Reply:
Yes, the test ran on Kubernetes with Kubernetes network and virtualization constraints and thus the numbers are applicable to it.